sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java

```diff
@@ -182,65 +272,41 @@ String getEndpoint() {
-    }
-
-    static void validateAccountProperties(AccountProperties expectedAccountProperties, AccountProperties actualAccountProperties) {
-        assertEquals(expectedAccountProperties.getLimit(), actualAccountProperties.getLimit());
-        assertNotNull(actualAccountProperties.getCount());
-    }
-
-    private static void validateTrainingDocuments(List<TrainingDocumentInfo> expectedTrainingDocuments, List<TrainingDocumentInfo> actualTrainingDocuments) {
-        List<TrainingDocumentInfo> actualTrainingList = actualTrainingDocuments.stream().collect(Collectors.toList());
-        List<TrainingDocumentInfo> expectedTrainingList = expectedTrainingDocuments.stream().collect(Collectors.toList());
-        assertEquals(expectedTrainingList.size(), actualTrainingList.size());
-        for (int i = 0; i < actualTrainingList.size(); i++) {
-            TrainingDocumentInfo expectedTrainingDocument = expectedTrainingList.get(i);
-            TrainingDocumentInfo actualTrainingDocument = actualTrainingList.get(i);
-            assertEquals(expectedTrainingDocument.getName(), actualTrainingDocument.getName());
-            assertEquals(expectedTrainingDocument.getPageCount(), actualTrainingDocument.getPageCount());
-            assertEquals(expectedTrainingDocument.getTrainingStatus(), actualTrainingDocument.getTrainingStatus());
-            validateErrors(expectedTrainingDocument.getDocumentErrors(), actualTrainingDocument.getDocumentErrors());
-        }
-    }
-
-    private static void validateErrors(List<FormRecognizerError> expectedErrors, List<FormRecognizerError> actualErrors) {
-        if (expectedErrors != null && actualErrors != null) {
-            List<FormRecognizerError> actualErrorList = actualErrors.stream().collect(Collectors.toList());
-            List<FormRecognizerError> expectedErrorList = expectedErrors.stream().collect(Collectors.toList());
-            assertEquals(expectedErrorList.size(), actualErrorList.size());
-            for (int i = 0; i < actualErrorList.size(); i++) {
-                FormRecognizerError expectedError = expectedErrorList.get(i);
-                FormRecognizerError actualError = actualErrorList.get(i);
-                assertEquals(expectedError.getCode(), actualError.getCode());
-                assertEquals(expectedError.getMessage(), actualError.getMessage());
+    private String createStorageAndGenerateSas(String folderPath) {
+        if (interceptorManager.isPlaybackMode()) {
+            return "https://isPlaybackmode";
+        } else {
+            String accountName = Configuration.getGlobalConfiguration().get("PRIMARY_STORAGE_ACCOUNT_NAME");
+            String accountKey = Configuration.getGlobalConfiguration().get("PRIMARY_STORAGE_ACCOUNT_KEY");
+            StorageSharedKeyCredential credential = new StorageSharedKeyCredential(accountName, accountKey);
+            String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
```

Comment:
`PRIMARY_STORAGE_ACCOUNT_NAME` should be something like `PRIMARY_STORAGE_ACCOUNT_URL` or this hardcoded URL (`https://%s.blob.core.windows.net`) will affect testing in other clouds.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10380